File: frontend/src/types/analysis.ts
export interface MetricResult {
  metric_name: string;
  metric_type: 'primary' | 'secondary' | 'guardrail';
  control_mean: number;
  treatment_mean: number;
  absolute_effect: number;
  relative_effect: number;
  ci_lower: number;
  ci_upper: number;
  p_value: number;
  is_significant: boolean;
  sample_size_control: number;
  sample_size_treatment: number;
}

export interface ExperimentResults {
  experiment_id: string;
  experiment_name: string;
  status: string;
  analysis_type: string;
  metrics: MetricResult[];
  recommendation: 'ship' | 'dont_ship' | 'keep_running' | 'inconclusive';
  computed_at: string;
}

export interface PowerCurvePoint {
  effect_size: number;
  power: number;
}

export interface SampleSizeRequest {
  baseline_rate: number;
  minimum_detectable_effect: number;
  alpha?: number;
  power?: number;
  metric_type?: string;
  variance?: number;
  daily_traffic?: number;
  n_variants?: number;
}

export interface SampleSizeResponse {
  sample_size_per_variant: number;
  total_sample_size: number;
  estimated_days: number | null;
  power: number;
  alpha: number;
  minimum_detectable_effect: number;
  baseline_rate: number;
  power_curve: PowerCurvePoint[];
}

File: backend/app/schemas/analysis.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class MetricResult(BaseModel):
    metric_name: str
    metric_type: str  # primary|secondary|guardrail
    control_mean: float
    treatment_mean: float
    absolute_effect: float
    relative_effect: float  # percentage
    ci_lower: float
    ci_upper: float
    p_value: float
    is_significant: bool
    sample_size_control: int
    sample_size_treatment: int


class ExperimentResults(BaseModel):
    experiment_id: uuid.UUID
    experiment_name: str
    status: str
    analysis_type: str
    metrics: list[MetricResult]
    recommendation: str  # "ship" | "dont_ship" | "keep_running" | "inconclusive"
    computed_at: datetime
